### ISSUE TITLE
failed to set ip

### ISSUE THREAD

**Issue by neuroscr** [loki-project/loki-network#97]
Not crashing any more but not getting any further:

Running as root
```
[NFO] unnamed (166) Tue Dec  4 02:38:04 2018 UTC llarp/context.cpp:130	lokinet-0.3.1-3d0f28b (dev build)
[NFO] unnamed (166) Tue Dec  4 02:38:04 2018 UTC llarp/context.cpp:131	starting up
[NFO] unnamed (166) Tue Dec  4 02:38:04 2018 UTC llarp/context.cpp:95	nodedb_dir loaded 0 RCs from [/root/.lokinet/netdb]
[NFO] llarpdev-main (166) Tue Dec  4 02:38:04 2018 UTC llarp/router.cpp:1304	add upstream resolver 128.52.130.209
[NFO] llarpdev-main (166) Tue Dec  4 02:38:04 2018 UTC llarp/router.cpp:1304	add upstream resolver 85.208.208.141
[NFO] llarpdev-main (166) Tue Dec  4 02:38:04 2018 UTC llarp/router.cpp:1304	add upstream resolver 103.236.162.119
[NFO] llarpdev-main (166) Tue Dec  4 02:38:04 2018 UTC llarp/router.cpp:1309	set local dns to 127.3.2.1:53
[NFO] llarpdev-main (166) Tue Dec  4 02:38:04 2018 UTC llarp/router.cpp:998	outbound link ready
[NFO] llarpdev-main (166) Tue Dec  4 02:38:04 2018 UTC llarp/router.cpp:771	Bound RPC server to 127.0.0.1:1190
[NFO] llarpdev-main (166) Tue Dec  4 02:38:04 2018 UTC llarp/router.cpp:786	You have 1 inbound links
[NFO] llarpdev-main (166) Tue Dec  4 02:38:04 2018 UTC llarp/router.cpp:799	Loading Addr: 206.81.100.174:1090 into our RC
[NFO] llarpdev-main (166) Tue Dec  4 02:38:04 2018 UTC llarp/router.cpp:838	Your Encryption pubkey 8458adaa29c5220003b9f80f872379dbcd6008d04ad8b7d257bcbec357454524
[NFO] llarpdev-main (166) Tue Dec  4 02:38:04 2018 UTC llarp/router.cpp:841	Your Identity pubkey 608df917988b25a0b1b36365053363328d765a53f1cf48cbf64b1c19f537cdd3
[NFO] llarpdev-main (166) Tue Dec  4 02:38:04 2018 UTC llarp/router.cpp:852	Signing rc...
[NFO] llarpdev-main (166) Tue Dec  4 02:38:04 2018 UTC llarp/router.cpp:864	have 0 routers
[NFO] llarpdev-main (166) Tue Dec  4 02:38:04 2018 UTC llarp/router.cpp:934	accepting transit traffic
[WRN] llarpdev-main (166) Tue Dec  4 02:38:04 2018 UTC vendor/libtuntap-master/tuntap-unix.c:354	Can't unset debug
[NFO] llarpdev-main (166) Tue Dec  4 02:38:04 2018 UTC vendor/libtuntap-master/tuntap-unix-linux.c:197	IPv6 is not implemented on your system
[WRN] llarpdev-main (166) Tue Dec  4 02:38:04 2018 UTC llarp/ev.cpp:269	failed to set ip
[ERR] llarpdev-main (166) Tue Dec  4 02:38:04 2018 UTC llarp/router.cpp:892	Failed to initialize service node
```

lokinet.ini
```
# this configuration was auto generated with 'sane' defaults
# change these values as desired


# router settings block
[router]
# uncomment these to manually set public address and port
# this is required on providers like AWS because of their firewall rules
# public-address=your.ip.goes.here
# public-port=1090

# number of crypto worker threads 
threads=4
# path to store signed RC
contact-file=/root/.lokinet/self.signed
# path to store transport private key
transport-privkey=/root/.lokinet/transport.private
# path to store identity signing key
identity-privkey=/root/.lokinet/identity.private
# path to store signed RC
contact-file=/root/.lokinet/self.signed

# uncomment following line to set router nickname to 'lokinet'
nickname=llarpdev-main


# system settings for priviledges and such
[system]
#user=lokinet
#group=lokinet
user=root
group=root

# dns provider configuration section
[dns]
# opennic us resolver
upstream=128.52.130.209
# opennic eu resolver
upstream=85.208.208.141
# opennic au resolver
upstream=103.236.162.119
bind=127.3.2.1:53


# network database settings block 
[netdb]
# directory for network database skiplist storage
dir=/root/.lokinet/netdb


# publish network interfaces for handling inbound traffic
[bind]
eth0=1090
```

I can provide SSH if you'd like.

**Comment by neuroscr** [comment]
figured out -v does debug mode, so here's a more detailed log

```
[DBG] unnamed (397) Tue Dec  4 04:18:50 2018 UTC daemon/main.cpp:74	debug logging activated
[NFO] unnamed (397) Tue Dec  4 04:18:50 2018 UTC llarp/context.cpp:130	lokinet-0.3.1-3d0f28b (dev build)
[NFO] unnamed (397) Tue Dec  4 04:18:50 2018 UTC llarp/context.cpp:131	starting up
[NFO] unnamed (397) Tue Dec  4 04:18:50 2018 UTC llarp/context.cpp:95	nodedb_dir loaded 0 RCs from [/root/.lokinet/netdb]
[NFO] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC llarp/router.cpp:1304	add upstream resolver 128.52.130.209
[NFO] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC llarp/router.cpp:1304	add upstream resolver 85.208.208.141
[NFO] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC llarp/router.cpp:1304	add upstream resolver 103.236.162.119
[NFO] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC llarp/router.cpp:1309	set local dns to 127.3.2.1:53
[DBG] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC llarp/ev.cpp:494	bind to 206.81.100.174:1090
[DBG] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC llarp/ev.cpp:494	bind to 0.0.0.0:0
[NFO] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC llarp/router.cpp:998	outbound link ready
[DBG] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC llarp/router.cpp:1132	find or create encryption.key
[DBG] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC llarp/router.cpp:1104	find or create identity.key
[NFO] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC llarp/router.cpp:771	Bound RPC server to 127.0.0.1:1190
[NFO] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC llarp/router.cpp:786	You have 1 inbound links
[NFO] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC llarp/router.cpp:799	Loading Addr: 206.81.100.174:1090 into our RC
[NFO] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC llarp/router.cpp:838	Your Encryption pubkey 8458adaa29c5220003b9f80f872379dbcd6008d04ad8b7d257bcbec357454524
[NFO] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC llarp/router.cpp:841	Your Identity pubkey 608df917988b25a0b1b36365053363328d765a53f1cf48cbf64b1c19f537cdd3
[NFO] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC llarp/router.cpp:852	Signing rc...
[DBG] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC llarp/router.cpp:325	verify RC signature
[NFO] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC llarp/router.cpp:864	have 0 routers
[DBG] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC llarp/router.cpp:866	starting outbound link
[DBG] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC llarp/router.cpp:880	Link utp started
[NFO] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC llarp/router.cpp:934	accepting transit traffic
[DBG] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC llarp/ev.cpp:253	set ifname to 
[WRN] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC vendor/libtuntap-master/tuntap-unix.c:354	Can't unset debug
[NFO] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC vendor/libtuntap-master/tuntap-unix-linux.c:197	IPv6 is not implemented on your system
[WRN] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC llarp/ev.cpp:269	failed to set ip
[ERR] llarpdev-main (397) Tue Dec  4 04:18:50 2018 UTC llarp/router.cpp:892	Failed to initialize service node
```

**Comment by majestrate** [comment]
`ifname` is not set and it's trying to set it regardless, it's bug.

**Comment by KeeJef** [comment]
getting same behavior on one of my routers

**Comment by neuroscr** [comment]
this is fixed in #230
